MatrixSDK/JSONModels/MXEvent.m

```diff
@@ -668,20 +673,16 @@ - (MXEvent*)editedEventFromReplacementEvent:(MXEvent*)replaceEvent
         // Reuse its decryption data
         replaceEventDecryptionResult = [replaceEvent decryptionResult];
     }
-    else if (event.content[kMXMessageBodyKey] && newContentDict && [newContentDict[kMXMessageTypeKey] isEqualToString:event.content[kMXMessageTypeKey]])
+    else if (newContentDict)
     {
         editedEventDict = [event.JSONDictionary mutableCopy];
-        NSMutableDictionary *editedEventContentDict = [editedEventDict[@"content"] mutableCopy];
-        editedEventContentDict[kMXMessageBodyKey] = newContentDict[kMXMessageBodyKey];
-        editedEventContentDict[@"formatted_body"] = newContentDict[@"formatted_body"];
-        editedEventContentDict[@"format"] = newContentDict[@"format"];
-        editedEventDict[@"content"] = editedEventContentDict;
+        editedEventDict[@"content"] = newContentDict;
```

Comment:
If I edit an event that is a reply, I wonder if this change would wipe out the reply relationship on the original? There might be other examples that should be preserved too.

Reply:
Good catch, I'll switch it to just simply check for poll start key content as well. 👍